### ISSUE TITLE
Yolov8 Transfer Learning to add custom Object Detection to pretrained -pose model

### ISSUE THREAD

**Issue by richardverheyen** [ultralytics/ultralytics#6087]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I'd like to detect players (persons) as well as balls for my sports app in the same detection run. 

Previously I've trained simple ball detection weights using data I've annotated on Roboflow, but instead of starting from scratch, I'd like to use either yolov8m-pose or yolov8m's coco weights, then (freeze?) the person detection layers, and add my ball detection training on at the end. The end result should be object detection yolov8 weights which detect ['person', 'ball'] as classes.

1. How do I identify that I want to keep the class of 'person' from the pretrained input weights? (whether sourced from detection or -pose).
2.  The 'freeze' config seems to be for "layers" not freezing classes, and 'pretrained' is set to True by default, do I set this as the same path as the model? The [docs](https://docs.ultralytics.com/usage/cfg/#train) for these confuse me.
3. Does my ball detection dataset need to have all instances of 'person' annotated in it as well?
5. Is there a performance downside to detecting lots of additional classes, (court features, scoreboards etc.) 
6. Can this be done through the CLI, or do I need a similar process to the [yolov5 example](https://docs.ultralytics.com/yolov5/tutorials/transfer_learning_with_frozen_layers/?h=freeze)

Thanks Ultralytics team, you're the best!

**Comment by glenn-jocher** [comment]
@richardverheyen dear User,

You're on the right track, great questions! Here's a brief guidance:

1. If the 'person' class is part of the pretrained weights, then when you reuse those weights, the ability to detect the 'person' class is implicitly retained. You do not need to explicitly specify this.

2. You're correct in assuming that 'freeze' refers to layers not classes. If 'freeze' is set to true, the specified layers of the model will not be updated during the training process. The pretrained weights will be used as they are. The 'pretrained' flag indicates whether or not to use pretrained weights to start training, and setting it to 'True' makes use of the pretrained weights.

3. Inclusion of 'person' instances in your ball detection dataset would be a good practice, but it's not absolutely necessary if you're reusing weights that already include a 'person' class. It could possibly help achieve more accurate detections in the context of your images. 

4. As for adding more classes like court features, scoreboards, etc., there can be a minor performance decrease due to increased model complexity, but on a well-optimized model like YOLOv8, the impact should be minimal.

5. Changing the classes and freezing certain layers is done via modifying the model configuration file. You would use this modified configuration file during training via the CLI, so it can indeed be accomplished through CLI. 

Hope this helps and good luck with your project! Your enthusiasm for object detection is much appreciated; the credit goes to our amazing YOLO community and the Ultralytics team for making this possible.

**Comment by richardverheyen** [comment]
Thanks for your help Glenn, unfortunately I still don't quite get it. 
I made a simplified, smaller version of my dataset in roboflow at 640x640, and I'm trying to train a yolov8s model to detect 'volleyball' and 'person', with the pretrained weights of 'person' being carried over from the default yolov8s weights, and 'volleyball' coming from my dataset. 

This is the train command I'm using (adjusted from the [docs](https://docs.ultralytics.com/tasks/detect/#train)  )
`!yolo task=detect mode=train data={dataset.location}/data.yaml model=yolov8s.yaml pretrained=yolov8s.pt epochs=25 imgsz=640 plots=True`

Here's my data.yaml file which i've adjusted based on my flawed understanding: 
`names:
- volleyball
- person #added manually to bring across person label from pretrained
nc: 2  #updated from 1 to 2
roboflow:
  license: CC BY 4.0
  project: beach-volleyball-640x640
  url: https://universe.roboflow.com/richard-verheyen-24xnm/beach-volleyball-640x640/dataset/1
  version: 1
  workspace: richard-verheyen-24xnm
test: ../test/images
train: Beach-Volleyball-640x640-1/train/images
val: Beach-Volleyball-640x640-1/valid/images
`
![image](https://github.com/ultralytics/ultralytics/assets/18094975/c35c7dcc-dede-4419-8e15-6c37bfb5cf9e)
![image](https://github.com/ultralytics/ultralytics/assets/18094975/d81685fe-2279-4108-9083-7ca2edc10e32)
![image](https://github.com/ultralytics/ultralytics/assets/18094975/58bdde30-a581-4c2e-a735-17e2cbad4c85)

What am I missing here to pull the trained 'person' weights over from weights pretrained on the coco dataset?
As always, thanks so much for your help with this. You guys are the best!
Rich

**Comment by glenn-jocher** [comment]
@richardverheyen hi Rich,

It seems like you are doing everything correct with your data.yaml and configuration. 

By specifying the yolov8s.pt (the pretrained weights file) in your train command, you should automatically get the 'person' class along with any other classes the yolov8s model was trained on. 

Remember that object detection models like YOLO do not have separate weights for each class, but rather, they share weights across classes. Thus, you don't have to explicitly pull out 'person' weights - they are already included in the pretrained weights file.

The 'names' field in the data.yaml does not need to list the 'person' class, provided the class is included in the pretrained model. It should list the classes that are contained in your dataset.

Please ensure that instances of 'person' in your images are well represented. Although you are using pretrained weights trained on diverse images, for optimal performance, the model needs to understand your specific context as well. 

However, the 'person' class in your training results will only display and compute metrics for the 'person' instances that the model found in your dataset. So if your dataset does not contain any 'person' annotations, then the results 'person' class will always show zero detections, zero labels etc, regardless of the pretrained weights used.

Let me know if you have any other questions. Keep up the good work!

Glenn

**Comment by Mayuresh999** [comment]
hello @richardverheyen have you found out anything in this? I am in a same situation where I want to train model on a custom dataset while keeping learnings about previous classes intact.
